contracts/hermez/future_versions/lib/HermezHelpersV2.sol

```diff
@@ -200,120 +200,190 @@ contract HermezHelpersV2 is Initializable {
+                let mlengthmod := mod(mlength, 32)
+                let submod := sub(32, slengthmod)
+                let mc := add(_postBytes, submod)
+                let end := add(_postBytes, mlength)
+                let mask := sub(exp(0x100, submod), 1)
+
+                sstore(sc, add(sload(sc), and(mload(mc), mask)))
+
+                for {
+                    sc := add(sc, 1)
+                    mc := add(mc, 0x20)
+                } lt(mc, end) {
+                    sc := add(sc, 1)
+                    mc := add(mc, 0x20)
+                } {
+                    sstore(sc, mload(mc))
                 }
-                default {
-                    // get the keccak hash to get the contents of the array
-                    mstore(0x0, _preBytes_slot)
-                    // Start copying to the last used word of the stored array.
-                    let sc := add(keccak256(0x0, 0x20), div(slength, 32))
-
-                    // save new length
-                    sstore(_preBytes_slot, add(mul(newlength, 2), 1))
-
-                    // Copy over the first `submod` bytes of the new data as in
-                    // case 1 above.
-                    let slengthmod := mod(slength, 32)
-                    let mlengthmod := mod(mlength, 32)
-                    let submod := sub(32, slengthmod)
-                    let mc := add(_postBytes, submod)
-                    let end := add(_postBytes, mlength)
-                    let mask := sub(exp(0x100, submod), 1)
-
-                    sstore(sc, add(sload(sc), and(mload(mc), mask)))
-
-                    for {
-                        sc := add(sc, 1)
-                        mc := add(mc, 0x20)
-                    } lt(mc, end) {
-                        sc := add(sc, 1)
-                        mc := add(mc, 0x20)
-                    } {
-                        sstore(sc, mload(mc))
-                    }
-
-                    mask := exp(0x100, sub(mc, end))
-
-                    sstore(sc, mul(div(mload(mc), mask), mask))
+
+                mask := exp(0x100, sub(mc, end))
+
+                sstore(sc, mul(div(mload(mc), mask), mask))
+            }
+        }
+    }
+
+    /**
+     * @dev Calculate the circuit input hashing all the elements
+     * @param sender tx sender
+     * @param stateRoot state root
+     * @param tokenIDs list token ids to withdraw
+     * @param amountWithdraws list amounts to withdraw
+     * @param idxs list idxs to withdraw
+     * @param nTokens number of tokens to withdraw
+     * @return circuit input
+     */
+    function _getInputWithdraw(
+        address sender,
+        uint256 stateRoot,
+        uint32[] memory tokenIDs,
+        uint192[] memory amountWithdraws,
+        uint48[] memory idxs,
+        uint256 nTokens
+    ) internal pure returns (uint256) {
+        bytes memory inputBytes;
+
+        uint256 ptr; // Position for writing the buffer
+
+        assembly {
+            let inputBytesLength := add(
```

Comment:
I don't know what's the difference of gas cost between doing it in assembly or in abi.encodePacket, i think it's worth it to check the comparative becasue it's much more readable, if the difference is to high maybe can be worth it doing it in assembly, but since the arrays only can be of 4 length i suspect that the difference won't be so much!.

Although, if you want to do it in assembly, i think the calldataCopy could be more efficient!